src/main/scala/edu/ie3/simona/agent/grid/GridAgent.scala

```diff
@@ -71,20 +70,28 @@ object GridAgent extends DBFSAlgorithm with DCMAlgorithm {
       simonaConfig: SimonaConfig,
       listener: Iterable[ActorRef[ResultEvent]],
   ): Behavior[Message] = Behaviors.withStash(100) { buffer =>
-    // this determines the agents regular time bin it wants to be triggered e.g. one hour
-    val resolution: Long = simonaConfig.simona.powerflow.resolution.toSeconds
+    val cfg = simonaConfig.simona
 
-    val simStartTime: ZonedDateTime = TimeUtil.withDefaults
-      .toZonedDateTime(simonaConfig.simona.time.startDateTime)
+    given simStartTime: ZonedDateTime = cfg.time.simStartTime
+    val simEndTime = cfg.time.simEndTime
+
+    // this determines the agents regular time bin it wants to be triggered e.g. one hour
+    // if no resolution is given, we use the end time to prevent powerflow calculation
+    val resolution: Long = cfg.powerflow match {
+      case Some(value) =>
+        value.resolution.toSeconds
+      case None =>
+        // powerflow is only performed after the simulation ended
+        simEndTime.toTick + 1
+    }
```

Comment:
Why do we perform a powerflow also in the case we don't want to perform a powerflow?

Reply:
That's my point. It looks a bit odd here. How about making resolution Option[Long]?